### PR TITLE
DP-840-sigint-handling

### PR DESCRIPTION
**Pull request by jeff2zhang**
ishell upstream source code Console exits on SIGINT, which is not what deepshell wanted. Added an exit_on_SIGINT flag to Console constructor, which is default to act exactly the same as the upstream but gives us options on how to handle SIGINT

### CODE REVIEW

**Comment by jeff2zhang** [comment]
The stock Python readline module seems to be sufficient.
The PyPI readline v6.2.4.1 package is discontinued and replaced by gnureadline.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 *.pyc
 *.log
 *.egg-info/
+
+.idea
```
Comment (zayfod) [inline .gitignore]: `.idea/` plz as this is for directories only.

File: ishell/console.py [hunk 2]
```diff
@@ -7,11 +7,12 @@
 
 
 class Console(object):
-    def __init__(self, prompt="Prompt", prompt_delim=">"):
+    def __init__(self, prompt="Prompt", prompt_delim=">", exit_on_SIGINT=True):
```
Comment (zayfod) [inline ishell/console.py]: please, use lower case for variables - `exit_on_sigint`.

**Review by zayfod — CHANGES_REQUESTED** [review]
Two minor changes, please.

**Review by zayfod — APPROVED** [review]

**Pull request merged**